src/Corax/IndexEntry/IndexEntryWriter.cs

```diff
@@ -91,236 +117,254 @@ public void WriteRaw(int field, ReadOnlySpan<byte> binaryValue)
+
+        if (FreeSpace < sizeof(IndexEntryFieldType) + value.Length + 4 * sizeof(long))
+            UnlikelyGrowAuxiliaryBuffer();
 
         int dataLocation = _dataIndex;
+        var buffer = Buffer;        
 
         // Write known field pointer.
-        ref int fieldLocation = ref _knownFieldsLocations[field];
+        ref int fieldLocation = ref KnownFieldsLocations[field];
         fieldLocation = dataLocation | Constants.IndexWriter.IntKnownFieldMask;
 
         // Write the tuple information. 
-        fixed (byte* bufferPtr = _buffer)
-        {
-            ref var indexEntryField = ref Unsafe.AsRef<IndexEntryFieldType>(bufferPtr + dataLocation);
-
-            indexEntryField = IndexEntryFieldType.Tuple;
-            dataLocation += sizeof(IndexEntryFieldType);
+        ref var indexEntryField = ref Unsafe.AsRef<IndexEntryFieldType>(Unsafe.AsPointer(ref buffer[dataLocation]));
+        indexEntryField = IndexEntryFieldType.Tuple;
+        dataLocation += sizeof(IndexEntryFieldType);
 
-            dataLocation += VariableSizeEncoding.Write(_buffer, longValue, dataLocation);
-            Unsafe.WriteUnaligned(ref _buffer[dataLocation], doubleValue);
-            dataLocation += sizeof(double);
-            dataLocation += VariableSizeEncoding.Write(_buffer, value.Length, dataLocation);
+        dataLocation += VariableSizeEncoding.Write(buffer, longValue, dataLocation);
+        Unsafe.WriteUnaligned(ref buffer[dataLocation], doubleValue);
+        dataLocation += sizeof(double);
+        dataLocation += VariableSizeEncoding.Write(buffer, value.Length, dataLocation);
 
-            // Copy the actual string data. 
-            if (value.Length != 0)
-            {
-                value.CopyTo(_buffer.Slice(dataLocation, value.Length));
-                dataLocation += value.Length;
-            }
-            else
-            {
-                // Signal that we will have to deal with the empties.
-                indexEntryField |= IndexEntryFieldType.Empty;
-            }
-
-            _dataIndex = dataLocation;
+        // Copy the actual string data. 
+        if (value.Length != 0)
+        {
+            value.CopyTo(buffer.Slice(dataLocation, value.Length));
+            dataLocation += value.Length;
+        }
+        else
+        {
+            // Signal that we will have to deal with the empties.
+            indexEntryField |= IndexEntryFieldType.Empty;
         }
+
+        _dataIndex = dataLocation;
     }
 
-    public void Write<TEnumerator>(int field, TEnumerator values, IndexEntryFieldType type = IndexEntryFieldType.Null)
+    public void Write<TEnumerator>(int field, TEnumerator values, IndexEntryFieldType type = IndexEntryFieldType.Null) 
         where TEnumerator : IReadOnlySpanIndexer
     {
         Debug.Assert(field < _knownFields.Count);
-        Debug.Assert(_knownFieldsLocations[field] == Invalid);
+        Debug.Assert(KnownFieldsLocations[field] == Invalid);
```

Comment:
Can you please add messages to both `Debug.Asserts`? Otherwise it's sometimes hard to figure out which one has failed. Especially when there is one next to another. In general it would be good to review the assertions in Corax and add the messages where it's missing.